### PR TITLE
Version 6.7.2 gas limit fix

### PR DESCRIPTION
**Pull request by danjm**

### CODE REVIEW

File: ui/app/components/app/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js [hunk 1]
```diff
@@ -8,6 +8,21 @@ export default class AdvancedTabContent extends Component {
...
   }
 
+  constructor (props) {
+    super(props)
+    this.state = {
+      gasPrice: this.props.customGasPrice,
+      gasLimit: this.props.customGasLimit,
+    }
+    this.changeGasPrice = debounce(this.changeGasPrice, 500)
+    this.changeGasLimit = debounce(this.changeGasLimit, 500)
```
Comment (whymarrh) [inline ui/app/components/app/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js]: Can we use the property syntax for the debounce? That is, wrap the property value in the debounce call directly. If we can remove theses lines we can move the state to a property as well and remove the constructor.

File: ui/app/pages/confirm-token-transaction-base/confirm-token-transaction-base.container.js [hunk 2]
```diff
@@ -14,15 +17,21 @@ import {
...
   const {
-    txData: { txParams: { to: tokenAddress, data } = {} } = {},
-    fiatTransactionTotal,
-    ethTransactionTotal,
+    txData: { id: transactionId, txParams: { to: tokenAddress, data } = {} } = {},
   } = confirmTransaction
 
-
+  const transaction = selectedAddressTxList.find(({ id }) => id === (Number(paramsTransactionId) || transactionId))
+  console.log('!!! * transaction', transaction)
```
Comment (whymarrh) [inline ui/app/pages/confirm-token-transaction-base/confirm-token-transaction-base.container.js]: Nit: console log

File: ui/app/pages/confirm-transaction-base/confirm-transaction-base.container.js [hunk 3]
```diff
@@ -32,50 +33,37 @@ const mapStateToProps = (state, ownProps) => {
...
+  const transaction = R.find(({ id }) => id === (transactionId || Number(paramsTransactionId)))(selectedAddressTxList)
   const {
     from: fromAddress,
     to: txParamsToAddress,
     gasPrice,
     gas: gasLimit,
     value: amount,
     data,
-  } = txParams
+  } = transaction && transaction.txParams || txParams
```
Comment (whymarrh) [inline ui/app/pages/confirm-transaction-base/confirm-transaction-base.container.js]: Can we express this instead with a ternary?

**Review by whymarrh — CHANGES_REQUESTED** [review]
A few small comments

**Comment by whymarrh** [comment]
I'm testing out 8e967aa locally now

File: ui/app/components/app/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js [hunk 4]
```diff
@@ -20,15 +31,40 @@ export default class AdvancedTabContent extends Component {
...
+      this.setState({ gasLimit: 21000 })
       this.props.updateCustomGasLimit(21000)
+    } else {
+      this.props.updateCustomGasLimit(Number(e.target.value))
     }
-  }, 1000, { trailing: true })
+  }
+
+  onChangeGasPrice = (e) => {
+    this.setState({ gasPrice: e.target.value })
```
Comment (Gudahtt) [inline ui/app/components/app/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js]: Should `e.target.value` be converted to a `Number` here?

Edit: Likewise for `onChangeGasLimit` above

**Review by Gudahtt — CHANGES_REQUESTED** [review]

**Review by tmashuang — APPROVED** [review]

**Pull request merged**